cdap-support-bundle/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -105,4 +139,130 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
+    throws Exception {
+    SupportBundleEntityId bundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(bundleEntityId, StandardPermission.DELETE);
+    File baseDirectory = new File(cConf.get(Constants.SupportBundle.LOCAL_DATA_DIR));
+    File uuidFile = new File(baseDirectory, uuid);
+    if (!baseDirectory.exists()) {
+      throw new NotFoundException("No content in Support Bundle.");
+    }
+    if (!uuidFile.exists()) {
+      throw new NotFoundException(String.format("No such uuid '%s' in Support Bundle.", uuid));
+    }
+    bundleGenerator.deleteOldFoldersIfExceedLimit(uuidFile);
+    responder.sendString(HttpResponseStatus.OK, String.format("Successfully deleted bundle %s", uuid));
+  }
+
+  /**
+   * Exports all selected files as a ZIP archive file.
+   *
+   * @param uuid the bundle id which is also the uuid fileJson body
+   */
+  @POST
+  @Path("/support/bundle/{uuid}/download")
+  public void appsExport(FullHttpRequest request, HttpResponder responder, @PathParam("uuid") String uuid)
+    throws Exception {
+    SupportBundleEntityId bundleEntityId = new SupportBundleEntityId(uuid);
+    contextAccessEnforcer.enforce(bundleEntityId, StandardPermission.GET);
+    String requestContent = request.content().toString(StandardCharsets.UTF_8);
+    if (requestContent == null || requestContent.length() == 0) {
+      throw new BadRequestException("Request body is empty.");
+    }
+    SupportBundleExportRequest bundleExportRequest =
+      GSON.fromJson(requestContent, SupportBundleExportRequest.class);
+    File tempDir = new File(cConf.get(Constants.SupportBundle.LOCAL_DATA_DIR),
+                            cConf.get(Constants.SupportBundle.SUPPORT_BUNDLE_TEMP_DIR)).getAbsoluteFile();
+    DirUtils.mkdirs(tempDir);
+    java.nio.file.Path tmpPath = Files.createTempFile(tempDir.toPath(),
+                                                      String.format(CONTENT_DISPOSITION_VALUE,
+                                                                    System.currentTimeMillis()),
+                                                      ".zip");
+    List<String> requestFiles = bundleExportRequest.getSupportBundleRequestFileList().getFiles();
+    File uuidFile = new File(cConf.get(Constants.SupportBundle.LOCAL_DATA_DIR), uuid);
+    if (!uuidFile.exists()) {
+      throw new NotFoundException(String.format("This bundle id %s is not existed", uuid));
+    }
+    try {
+      MessageDigest digest = MessageDigest.getInstance("SHA-256");
+      try (ZipOutputStream zipOut = new ZipOutputStream(
+        new DigestOutputStream(Files.newOutputStream(tmpPath, StandardOpenOption.TRUNCATE_EXISTING), digest))) {
+        for (String filePath : requestFiles) {
+          File requestFile = new File(uuidFile, filePath);
+          if (requestFile.exists()) {
+            ZipEntry entry = new ZipEntry(uuidFile.getName() + File.separator + filePath);
```

Comment:
Zip entry must use `/` as the path separator (https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT section 4.4.17)